src/components/FormContext.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { FC, createContext, useContext } from 'react'
+import { useForm, FieldValues } from 'react-hook-form'
+
+import { FormContextProps, FormValuesWithSchema } from './types'
+
+export const InternalFormContext = createContext<FormValuesWithSchema<
+  FieldValues
+> | null>(null)
+
+export function useFormContext<T extends FieldValues>(): FormValuesWithSchema<
+  T
+> {
+  return useContext(InternalFormContext) as FormValuesWithSchema<T>
+}
+
+export const FormContext: FC<FormContextProps> = props => {
```

Comment:
```suggestion
export const FormContext: FC<FormContextProps> = props => {
  const { mode = ValidationMode.onSubmit } = props
```

Reply:
Unfortunatelly this is not possible, because ValidationMode is an interface, not an enum :(

Reply:
And why not make it as a enum?

Reply:
I wanted to conform to the `react-hook-form` parameters, I just use what was defined in that lib, but I think I can change that